src/HotChocolate/Core/src/Abstractions/Execution/IQueryRequestBuilder.cs

```diff
@@ -44,27 +45,97 @@ public interface IQueryRequestBuilder
     IQueryRequestBuilder SetInitialValue(
         object? initialValue);
 
+    [Obsolete("Use `InitializeGlobalState`")]
     IQueryRequestBuilder SetProperties(
         Dictionary<string, object?>? properties);
 
+    /// <summary>
+    /// Initializes the global state of the request to the given
+    /// <paramref name="initialState" />.
+    /// </summary>
+    /// <param name="initialState">The initial state.</param>
+    /// <returns>The query request builder.</returns>
+    IQueryRequestBuilder InitializeGlobalState(
+        Dictionary<string, object?>? initialState);
+
+    [Obsolete("Use `InitializeGlobalState`")]
```

Comment:
Awesome :) was my first thought in this... if we clean this up we need to leave the old version as obsolete.